test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -336,8 +336,17 @@ public HubConnectionTests(ServerFixture<Startup> serverFixture, ITestOutputHelpe
                     await channel.WaitToReadAsync().AsTask().OrTimeout();
                     cts.Cancel();
 
-                    var results = await channel.ReadAllAsync().OrTimeout();
+                    List<int> results = new List<int>();
+                    try
+                    {
+                        await channel.ReadAllIntoListAsync(results).OrTimeout();
+                    }
+                    catch (OperationCanceledException)
```

Comment:
This fixes the unit test but I don't know if it is correct. Is reading from a cancelled channel throwing a `OperationCanceledException` correct?

Reply:
My understanding is yes, it is. A channel can be completed with an exception and it's expected that the channel will throw that exception after reading the last item written prior to completion.

Reply:
Looks like `ReadAllIntoListAsync` explicitly manifests the `Completion` task's error, which would be expected to yield this exception.